### ISSUE TITLE
你好，我想请问读取h5数据集的时候，如何打乱数据集呢？

### ISSUE THREAD

**Issue by stxg13056** [liangjiandeng/DLPan-Toolbox#6]
我自己训练模型，发现代码读取数据之后，貌似并没有进行shuffle，这导致训练结果会过拟合。

**Comment by XiaoXiao-Woo** [comment]
In fact, the data has been shuffled and the dataset class receives random indexes generated from the data loader. Therefore, there is no need to shuffle the data again.

**Comment by XiaoXiao-Woo** [comment]
In fact, the data has been shuffled and the dataset class receives random
indexes generated from the data loader. Therefore, there is no need to
shuffle the data again.

**Comment by liangjiandeng** [comment]
> 我自己训练模型，发现代码读取数据之后，貌似并没有进行shuffle，这导致训练结果会过拟合。

To ensure a fair comparison of all methods, we fix the data preprocessing and give the fixed dataset. You may adjust your network to solve the overfitting problem in your model. You can use the publicly datasets for pansharpening -- PanCollection. https://github.com/liangjiandeng/PanCollection

**Comment by stxg13056** [comment]
> In fact, the data has been shuffled and the dataset class receives random indexes generated from the data loader. Therefore, there is no need to shuffle the data again.

I tried to repeatedly train my model three times, using the default setting of output loss every 50 batches. However, the value of each repetition is the same, so I guess it is because the data is not shuffled everytime. The dataset I use is the dataset you provide.

**Comment by liangjiandeng** [comment]
> > In fact, the data has been shuffled and the dataset class receives random indexes generated from the data loader. Therefore, there is no need to shuffle the data again.
> 
> I tried to repeatedly train my model three times, using the default setting of output loss every 50 batches. However, the value of each repetition is the same, so I guess it is because the data is not shuffled everytime. The dataset I use is the dataset you provide.

Do you use the framework of the DLPan-Toolbox? see: https://github.com/liangjiandeng/DLPan-Toolbox/tree/main/01-DL-toolbox(Pytorch)

**Comment by stxg13056** [comment]
> > > In fact, the data has been shuffled and the dataset class receives random indexes generated from the data loader. Therefore, there is no need to shuffle the data again.
> > 
> > 
> > I tried to repeatedly train my model three times, using the default setting of output loss every 50 batches. However, the value of each repetition is the same, so I guess it is because the data is not shuffled everytime. The dataset I use is the dataset you provide.
> 
> Do you use the framework of the DLPan-Toolbox? see: https://github.com/liangjiandeng/DLPan-Toolbox/tree/main/01-DL-toolbox(Pytorch)

Yes，I use this framework to build my model, and use your dataset to train my model.

**Comment by XiaoXiao-Woo** [comment]
Show your training process if you can. As mentioned by [liangjiandeng](https://github.com/liangjiandeng), I have fixed the seed to guarantee reproducibility of results. I don't know if you're talking about repeating the training is the same or the iterations in the epoch are the same

**Comment by stxg13056** [comment]
![QQ图片20221111181806](https://user-images.githubusercontent.com/81797325/201319474-ff99f933-0125-4587-b095-692418a1a511.png)
To put it simply, every time I start training, the first two sets of output loss value are shown in the figure above. Then,  when I stop training and start it again, the first two sets of output loss values will be the same.
Theoretically, the second values should be different, right?

**Comment by XiaoXiao-Woo** [comment]
> ![QQ图片20221111181806](https://user-images.githubusercontent.com/81797325/201319474-ff99f933-0125-4587-b095-692418a1a511.png) To put it simply, every time I start training, the first two sets of output loss value are shown in the figure above. Then, when I stop training and start it again, the first two sets of output loss values will be the same. Theoretically, the second values should be different, right?

I think you're talking about repeating the training is the same.  Any time when you start training should be the same, which is reproducibility. This is no problem.

**Comment by stxg13056** [comment]
> 
Thanks for your advice. I may have problems with my own model. I will try again to modify it carefully.
